api/app/Services/SocialAccount/Google.php

```diff
@@ -37,15 +47,33 @@ public function service(string $service)
         return new $classname($this->client);
     }
 
-    public function auth(string $code = null)
+    public function auth(string $code = null, string $state = null)
     {
         if (!$code) {
             return $this->createAuthUrl();
         } else {
             $this->authenticate($code);
+            $user = $this->decodeUser($state);
+
             $token = $this->getAccessToken();
 
-            return '';
+            if (!$token) {
+                throw new GoogleOauthException();
+            }
+
+            $socialAccount = $this->socialAccountRepository->findByProvider(
+                SocialAccount::GOOGLE_SERVICE_ID,
+                $user->userId
+            );
+
+            $socialAccount->account_id = $user->userId;
+            $socialAccount->token = $token;
+            $socialAccount->refresh_token = $token['refresh_token'];
+            $socialAccount->expires_in = '2020-08-27 22:57:59';
```

Comment:
Why the value is hardcoded?

Reply:
Just temporary solution cause we need to decide do we need this field at all, if yep I guess better migrate it to nullable and switch to int from timestamp. Google token with expired value stores in token field in serialized json format. And google  also return expire value as int instead of timestamp